docs/source/key_components/rl_toolkit.rst

```diff
@@ -58,138 +54,115 @@ easily turned into a roll-out worker or an actor for synchronous and asynchronou
-
-      @abstractmethod
-      def __call__(self, state):
-          """Select an action based on a state."""
-          raise NotImplementedError
+``Policy`` is the most important concept in reinforcement learning. In MARO, the highest level abstraction of a policy
+object is ``AbsPolicy``. It defines the interface ``get_actions()`` which takes a batch of states a inputs and returns
+corresponding actions.
 
+The simplest type of policy is ``RuleBasedPolicy`` which generates actions by pre-defined rules. ``RuleBasedPolicy``
+is mostly used in naive scenarios. However, in most cases, we need to train the policy by interacting with the
+environment, these are the cases we need to use ``RLPolicy``. In MARO's design, a policy cannot train itself. Instead,
+polices could only be trained by :ref:`trainer` (we will introduce trainer later in this page). Therefore, in addition
+to ``get_actions``, ``RLPolicy`` also has a set of training-related interfaces, such as ``step()``, ``get_gradients()``
+and ``set_gradients()``. These interfaces will be called by trainers for training. As you may noticed, currently
+we assume policies are built upon deep learning models, so the training-related interfaces are specifically
+designed for gradient decent.
 
-  class RLPolicy(AbsPolicy):     
-      ...
+``RLPolicy`` is further divided into three types:
+- ``ValueBasedPolicy``: For valued-based policies.
+- ``DiscretePolicyGradient``: For gradient-based policies that generate discrete actions.
+- ``ContinuousPolicyGradient``: For gradient-based policies that generate continuous actions.
 
-      def record(self, key: str, state, action, reward, next_state, terminal: bool):
-          pass
+The above classes are all concrete classes. Users do not need to implement any new classes, but can directly
+create a policy object by configuring parameters. Here is a simple example:
 
-      def get_rollout_info(self):
-          pass
-
-      def get_batch_loss(self, batch: dict, explicit_grad: bool = False):
-          pass
-
-      def update(self, loss_info_list: List[dict]):
-          pass
+.. code-block:: python
 
-      def learn(self, batch: dict):
-          pass
+   ValueBasedPolicy(
+       name="policy",
+       q_net=MyQNet(state_dim=128, action_num=64),
+   )
 
-      def improve(self):
-          pass
 
+For now, you may have no idea about the ``q_net`` parameter, but don't worry, we will introduce it in the next section.
 
-.. _policy-manager:
+Model
+-----
 
-Policy Manager
---------------
+The above code snippet creates a ``ValueBasedPolicy`` object. Let's pay our attention to the parameter ``q_net``.
+``q_net`` accepts a ``DiscreteQNet`` object, and it serves as the core part of a ``ValueBasedPolicy`` object. In
+other words, ``q_net`` defines the model structure of the Q-network in the value-based policy, and further determines
+the policy's behavior. ``DiscreteQNet`` is an abstract class, and ``MyQNet`` is one of the user-defined implementation
```

Comment:
_one of the user-defined implementation_ -> _one of the user-defined implementation**s**_
or just 
_a user-defined implementation?_